src/futils.c

```diff
@@ -112,7 +112,7 @@ int git_futils_truncate(const char *path, int mode)
 	return 0;
 }
 
-git_off_t git_futils_filesize(git_file fd)
+int git_futils_filesize(git_object_size_t *out, git_file fd)
```

Comment:
Not sure how I feel about these. `off_t` directly maps to the mmap functions of the OS, using `git_object_size_t` feels weird to me

Reply:
Well, `mmap` does take an `off_t` but only for the offset into the file.  That doesn't correspond to the length of the file or memory-mapped segment.

I definitely don't think that we should use a signed type for file length, it simply can't be negative.  We wouldn't want to use `size_t` here, since on a 32-bit system, we want to be able to get the size of a file larger than 4 GB.

We could call this the more direct `uint64_t` here.

Reply:
Hum. Some platforms handle negative offsets in `mmap` just fine and treat it as valid. It's questionable though whether we're going to ever need that, especially considering that we'd have to fight with cross-platform behaviour here. So :+1: for `uint64_t`.

Reply:
Updated for `uint64_t`